src/mina-signer/signature.ts

```diff
@@ -204,3 +215,90 @@ function verify(
     return false;
   }
 }
+
+// legacy signatures
+
+/**
+ * Same as {@link sign}, but using the "legacy" style of hash input packing.
+ */
+function signLegacy(
```

Comment:
This feels like a lot of code duplication, since the signing algorithm between legacy and kimchi versions are equivalent.  They only call different versions of serialization, which could be encapsulated at a serialization layer.   This is just a note-- not saying it needs to be changed for this PR, but you might want to think about it.

Reply:
yeah I thought the same! but then I thought that this code is never ever supposed to change anyway, since we want to support the old signatures indefinitely. still would be nice to abstract out, I agree